test/inductor/test_max_autotune.py

```diff
@@ -221,6 +222,49 @@ def mm(a, b):
         with config.patch({"max_autotune": True}):
             torch.compile(mm, dynamic=dynamic)(a, b)
 
+    @parametrize("dynamic", (False, True))
+    def test_max_autotune_remote_caching(self, dynamic: bool):
+        from unittest.mock import patch
+
+        def mm(a, b):
+            a = torch.sin(a)
+            return a @ b
+
+        a = torch.randn(100, 10).cuda()
+        b = torch.randn(10, 100).cuda()
+
+        class MyCache:
+            def __init__(self, key, is_autotune=False):
+                self.cache = {}
+
+            def get(self, filenames):
+                return {
+                    file: self.cache[file] for file in filenames if file in self.cache
+                }
+
+            def put(self, filename, data):
+                self.cache[filename] = data
+
+        cache_module = (
+            "triton.runtime.fb_memcache.FbMemcacheRemoteCacheBackend"
+            if config.is_fbcode()
+            else "triton.runtime.cache.RedisRemoteCacheBackend"
+        )
+
+        with config.patch(
+            {
+                "max_autotune": True,
+                "use_autotune_local_cache": False,
+                "use_autotune_remote_cache": True,
+            }
+        ), patch.dict(os.environ), patch(cache_module, MyCache, create=True):
+            os.environ.pop("TRITON_CACHE_MANAGER", None)
+            torch.compile(mm, dynamic=dynamic)(a, b)
+            reset()
+            torch.compile(mm, dynamic=dynamic)(a, b)
+            reset()
+            torch.compile(mm, dynamic=dynamic)(a, b)
+
```

Comment:
Add an assert that we had a cache hit?

Reply:
So earlier today I was actually playing around with num_get and num_set in order to do exactly that, but it looks like `reset()` wasn't actually enough, the second/third time we end up skipping code generation, i assume there's some other cache i need to reset or disable. Any chance you know what it is off the top of your head?

Reply:
Perhaps it is the on-disk cache?

Reply:
Figured this out, will update the test.